packages/block-editor/src/components/url-input/index.js

```diff
@@ -60,13 +60,14 @@ class URLInput extends Component {
 
 		this.suggestionNodes = [];
 
-		this.isUpdatingSuggestions = false;
+		this.suggestionsRequest = null;
```

Comment:
These component instance properties are always a recipe for weird bugs since React doesn't usually observe changes in them. Maybe we should ideally get rid of them all?

Reply:
`this.suggestionNodes` is used to store refs, it can't be in state and needs to stay ref-like.

`this.suggestionsRequest` is used to prevent state updates from a request if another request has been issued in the meantime. It can't be in state because we need to set it to `null` on unmount, so that responses arriving after unmount are ignored.